### PR TITLE
Add automated snapshots for Jenkins data volume

### PR DESCRIPTION
**Pull request by lfdebrux**

### CODE REVIEW

**Comment by lfdebrux** [comment]
[Snapshot lifecycle policy (Jenkins auto backup)](https://trello.com/c/HDeShcyJ/524-snapshot-lifecycle-policy-jenkins-auto-backup)

File: terraform/modules/jenkins/jenkins/snapshots.tf [hunk 1]
```diff
@@ -0,0 +1,83 @@
+
+resource "aws_iam_role" "snapshot_jenkins_data_role" {
```
Comment (benvand) [inline terraform/modules/jenkins/jenkins/snapshots.tf]: We're hoping to move to all policy documents eventually. https://www.terraform.io/docs/providers/aws/d/iam_policy_document.html

Can we do these with policy docs?

Comment (lfdebrux) [reply terraform/modules/jenkins/jenkins/snapshots.tf]: Ooh let's try it

**Comment by benvand** [comment]
The `aws_sns_topic_policy` changes in the plan should be fine, however the `aws_route53_record` `allow_override` attribute has a recommendation to import the resource:
https://www.terraform.io/docs/providers/aws/guides/version-2-upgrade.html#resource-aws_route53_record

I've not got time to test this now but maybe tomorrow?

**Pull request merged**